### PR TITLE
[IotCentral] Add validation for new SKUs

### PR DESCRIPTION
**Pull request by kawayiboy**
[IotCentral] Add validation for new SKUs

## Description

Change sku type to utilize default enum validation
So the following powershell commend would result in following error message, while it would result in BadRequest previously

> New-AzIotCentralApp -ResourceGroupName IOTC -Name appame32 -Subdomain domainname145 -Sku T1
~~~
New-AzIotCentralApp : Cannot bind parameter 'Sku'. Cannot convert value "T1" to type
"Microsoft.Azure.Commands.IotCentral.Models.PSIotCentralAppSku". Error: "Unable to match the identifier name T1 to a
valid enumerator name. Specify one of the following enumerator names and try again:
S1, ST0, ST1, ST2"
At line:1 char:90
+ ... esourceGroupName IOTC -Name appame32 -Subdomain domainname145 -Sku T1
+                                                                        ~~
    + CategoryInfo          : InvalidArgument: (:) [New-AzIotCentralApp], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.Azure.Commands.Management.IotCentral.NewAzureRm
   IotCentralApp
~~~

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs [hunk 1]
```diff
@@ -112,7 +112,14 @@ public override void ExecuteCmdlet()
 
         private string GetAppSkuName()
         {
-            return this.Sku ?? PSIotCentralAppSku.S1.ToString();
+            string skuName;
+            if ((object)this.Sku != null && (int)this.Sku != 0) {
+                skuName = this.Sku.ToString();
+            } else {
+                skuName = PSIotCentralAppSku.S1.ToString();
```
Comment (r-delgadillo) [inline src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: This should default to "ST2" sku

Comment (poanchen) [reply src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: I will be changing the default sku for all the SDKs later on. so probably fine here. I think all of the SDKs for iotc are default to S1 right now.

File: src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs [hunk 2]
```diff
@@ -60,7 +60,7 @@ public class NewAzureRmIotCentralApp : IotCentralBaseCmdlet
             HelpMessage = "Pricing tier for IoT Central applications. Default value is S1.")]
         [PSArgumentCompleter("S1")]
         [ValidateNotNullOrEmpty]
-        public string Sku { get; set; }
+        public PSIotCentralAppSku Sku { get; set; }
```
Comment (markcowl) [inline src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: This is a breaking change, and ignores the argument completer - please add the new known values to the ArgumentCompleter, and implement the defaulting logic accordingly

Comment (kawayiboy) [reply src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: I've implemented default logic, and as argument completer would loop through the values of PSIotCentralAppSku right now, I just remove the PSArgumentCompleter

**Comment by kawayiboy** [comment]
@markcowl , the PR moved to here: https://github.com/Azure/azure-powershell/pull/11086

File: src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs (hunk 2 above)
Comment (kawayiboy) [reply src/IotCentral/IotCentral/IotCentral/NewAzureRmIotCentralApp.cs]: @markcowl , after discussion with our team, we think it's ok to have this "breaking change", when the customer install this new version of powershell, they would be notify to input correct value to create an IoTCentral app

**Pull request closed without merging**